src/Features/CSharp/Portable/CodeFixes/Nullable/CSharpPossiblyDeclareAsNullableCodeFixProvider.cs

```diff
@@ -0,0 +1,140 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Immutable;
+using System.Composition;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.CSharp.Extensions;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CSharp.CodeFixes.PossiblyDeclareAsNullable
+{
+    /// <summary>
+    /// If you apply a null test on a symbol that isn't nullable, then we'll help you make that symbol nullable.
+    /// For example: `nonNull == null`, `nonNull?.Property`
+    /// </summary>
+    [ExportCodeFixProvider(LanguageNames.CSharp, Name = PredefinedCodeFixProviderNames.PossiblyDeclareAsNullable), Shared]
+    internal class CSharpPossiblyDeclareAsNullableCodeFixProvider : CodeFixProvider
+    {
+        [ImportingConstructor]
+        public CSharpPossiblyDeclareAsNullableCodeFixProvider()
+        {
+        }
+
+        public sealed override ImmutableArray<string> FixableDiagnosticIds
+            => ImmutableArray.Create(IDEDiagnosticIds.PossiblyDeclareAsNullableDiagnosticId);
+
+        // No support for FixAll at the moment. For example, in public API code this fix is likely incorrect.
+        public override FixAllProvider GetFixAllProvider() => null;
+
+        public override async Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            var diagnostic = context.Diagnostics.First();
+            var document = context.Document;
+            var cancellationToken = context.CancellationToken;
+
+            var fixableType = await TryGetFixableTypeAsync(document, diagnostic, cancellationToken).ConfigureAwait(false);
+            if (fixableType is object)
+            {
+                context.RegisterCodeFix(new MyCodeAction(
+                    c => FixAsync(context.Document, diagnostic, c)),
+                    context.Diagnostics);
+            }
+        }
+
+        private async Task<TypeSyntax> TryGetFixableTypeAsync(Document document, Diagnostic diagnostic, CancellationToken cancellationToken)
+        {
+            var root = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+            var node = root.FindNode(diagnostic.Location.SourceSpan, getInnermostNodeForTie: true);
```

Comment:
`getInnermostNodeForTie` is very likely correct.  however, it's worthwihle to add a test to make sure we don't regress that.  the easiest is to take one of your use cases and put it in the argument of some invocatoin call.  This will ensure that we get the expression node and not the argument node.